### PR TITLE
Feat/transaction module crypto operations refactor part 1

### PR DESCRIPTION
**Pull request by xeno097**
### Changelog

This PR implements part of the refactored logic for interacting with the blockchain and the first blockchain based use case.

- Updates the `Dockerfile.testnet` to expose the forked node on all container ports.
- Adds a `docker-compose.dev.yml` file to make it easier running the testnet node when developing locally.
- Updates the default execution environment from `DEV` to `develop` to follow the branching strategy and adds the `ExecutionEnv` enum.
-  Adds the `NODE_ENV` field to the `AppConfig` struct (A value for this env variable has already been added in secrets manager)
- Defines the `IUseCase` interface which represents the interface that all the use cases must implement.
- Implements the `CryptoWithdrawalUseCase`
- Defines the `IBlockchainWalletProvider`, the `IBlockchainWallet` and their concrete implementations.
- Moves the newly created use cases in the transaction module into the `to-review` folder because they need to be improved/reimplemented as some might not work as they are currently implemented.
- Implements the `TestnetProvider` to trick the testnet by changing the USDC contract logic to allow us to mint USDC to an authenticated user account.
- implements the `BlockchainAddress` value object to describe a valid Ethereum address string.
- Defines a new transaction type `CRYPTO_WITHDRAWAL` to distinguish between bank and crypto withdrawals

### CODE REVIEW

**Review by pellerano — APPROVED** [review]

File: src/modules/transaction/infra/libs/aws/aws-blockchain-wallet.provider.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+      const command = new PutObjectCommand({
+        Bucket: this.config.privateKeysBycketName,
+        Key: `${id.inner}/private.pem`,
+        Body: encryptedPrivateKey,
+      });
+
+      await this.awsClient.s3Client.send(command);
+
+      const userWallet = new ethers.Wallet(newWallet.privateKey, this.blockchainProvider);
+      const alcanciaFaucetWallet = new ethers.Wallet(this.config.alcanciaFaucetPrivatekey, this.blockchainProvider);
```
Comment (DHurtado714-itesm) [inline src/modules/transaction/infra/libs/aws/aws-blockchain-wallet.provider.ts:89]: maybe we should put this on a separate method

File: src/modules/transaction/infra/libs/aws/aws-blockchain-wallet.provider.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    } catch (error: any) {
+      throw new BlockchainWalletProviderInternalError({
+        inner: error,
+        message: `Failed to create wallet for user with id ${id.inner}`,
+      });
+    }
+  }
+
+  // Private methods
+  private async _encrypt(privateKey: string): Promise<string> {
```
Comment (DHurtado714-itesm) [inline src/modules/transaction/infra/libs/aws/aws-blockchain-wallet.provider.ts:105]: maybe use an interface for the input

File: src/modules/transaction/core/infra/blockchain-wallet.provider.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+   */
+  balance(): number | Promise<number>;
+
+  /**
+   * Transfers the specified amount of the underlying token to the address in {@link TransferDto}.
+   */
+  transfer(dto: TransferDto): TransferResponseDto | Promise<TransferResponseDto>;
+}
+
+export class BlockchainWalletProviderInternalError extends BaseInternalError {}
```
[the remark this answers is not in the document]
Comment (xeno097) [reply src/modules/transaction/core/infra/blockchain-wallet.provider.ts:33]: Yup to namespace errors when they are logged so we know exactly from which module they have been thrown

File: src/modules/transaction/infra/libs/aws/aws-blockchain-wallet.provider.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+  alcanciaFaucetPrivatekey: string;
+  usdcContractAddress: string;
+}
+
+/**
+ * The value can be read {@link https://polygonscan.com/token/0x0000000000000000000000000000000000001010#readContract | here} by calling the decimals method.
+ *
+ * As we currently only support polyogn the value can be safely hardcoded.
+ */
+const MATIC_TOKEN_DECIMALS = 18;
```
[the remark this answers is not in the document]
Comment (xeno097) [reply src/modules/transaction/infra/libs/aws/aws-blockchain-wallet.provider.ts:27]: It is safe to hard code for the moment as this value doesn't change compared to other token contracts

**Pull request merged**